### PR TITLE
Semantic embedding

### PR DESCRIPTION
**Pull request by ahartel**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -137,6 +137,25 @@ result = model.embed(request)
...
+
+model = AlephAlphaModel(
+    AlephAlphaClient(host="https://api.aleph-alpha.com", token=os.getenv("AA_TOKEN")),
+    # You need to choose a model with multimodal capabilities for this example.
+    model_name = "luminous-base"
+)
+
+request = SemanticEmbeddingRequest(prompt=Prompt.from_text("This is an example."), type="symmetric")
+result = model.semantic_embed(request)
+
```
Comment (benbrandt) [inline README.md:155]: Add asymmetric example, doc + query

File: aleph_alpha_client/embedding.py [hunk 2]
```diff
@@ -56,3 +64,62 @@ def from_json(json: Dict[str, Any]) -> "EmbeddingResponse":
...
         )
+
+
+class SemanticRepresentation(Enum):
+    """
+    Available types of semantic representations that prompts can be embedded with.
+    """
+
+    Symmetric = "symmetric"
+    """
```
Comment (volkerstampa) [inline aleph_alpha_client/embedding.py:75]: ? I do not think there is pydoc for members, is there? Usually this is just added to the doc of the class.

File: aleph_alpha_client/embedding.py [hunk 3]
```diff
@@ -56,3 +64,62 @@ def from_json(json: Dict[str, Any]) -> "EmbeddingResponse":
...
+
+
+class SemanticEmbeddingResponse(NamedTuple):
+    model_version: str
+    embedding: List[float]
+
+    @staticmethod
+    def from_json(json: Dict[str, Any]) -> "SemanticEmbeddingResponse":
+        return SemanticEmbeddingResponse(
+            model_version=json["model_version"], embedding=json["embedding"]
```
Comment (volkerstampa) [inline aleph_alpha_client/embedding.py:124]: Alternatively `return SemanticEmbeddingResponse(**json)`

**Review by volkerstampa — CHANGES_REQUESTED** [review]
I propose a few minor changes.

**Pull request merged**